src/Monotonic.h

```diff
@@ -0,0 +1,26 @@
+#ifndef HALIDE_MONOTONIC_H
+#define HALIDE_MONOTONIC_H
+
+/** \file
+ *
+ * Methods for computing whether expressions are monotonic
+ */
+
+#include "IR.h"
+
+namespace Halide {
+namespace Internal {
+
+/**
+ * Detect whether an expression is monotonic increasing in a variable,
+ * decreasing, or unknown.
+ */
+enum MonotonicResult {Constant, MonotonicIncreasing, MonotonicDecreasing, Unknown};
```

Comment:
Done.

On Thu, Apr 21, 2016 at 5:19 PM, Dillon Sharlet <notifications@github.com>
wrote:

> In src/Monotonic.h
> <https://github.com/halide/Halide/pull/1152#discussion_r60675382>:
>
> > +
> > +/** \file
> > + *
> > + * Methods for computing whether expressions are monotonic
> > + */
> > +
> > +#include "IR.h"
> > +
> > +namespace Halide {
> > +namespace Internal {
> > +
> > +/**
> > + * Detect whether an expression is monotonic increasing in a variable,
> > + * decreasing, or unknown.
> > + */
> > +enum MonotonicResult {Constant, MonotonicIncreasing, MonotonicDecreasing, Unknown};
>
> enum class? (with rename of the enumeration values.)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/1152/files/b89e1d6e425474bf856b957049d0199820abac02#r60675382>
>